AudioKit/Common/Internals/CoreAudio/BufferedAudioUnit.h

```diff
@@ -0,0 +1,32 @@
+//
+//  BufferedAudioUnit.h
+//  AudioKit
+//
+//  Created by Dave O'Neill., revision history on Github.
+//  Copyright © 2018 AudioKit. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <AVFoundation/AVFoundation.h>
+#import <AudioToolbox/AudioToolbox.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void(^ProcessEventsBlock)(AudioBufferList        * _Nullable inBuffer,
+                                  AudioBufferList        * _Nonnull  outBuffer,
+                                  const AudioTimeStamp   * _Nonnull  timestamp,
+                                  AVAudioFrameCount                  frameCount,
+                                  const AURenderEvent    * _Nullable eventsListHead);
```

Comment:
If we wanted to future proof for multiple inputs we could change ProcessEventsBlock to something more like this:
```
typedef void(^ProcessEventsBlock)(
    AudioBufferList * _Nullable * _Nonnull  inBuffers,
    int                                     inBuffersCount,
    AudioBufferList             * _Nonnull  outBuffer,
    const AudioTimeStamp        * _Nonnull  timestamp,
    AVAudioFrameCount                       frameCount,
    const AURenderEvent         * _Nullable eventsListHead);
```

I'm just hesitant to do this now because we might end up with something entirely different once we actually implement it.